docs/source/notes/extending.rst

```diff
@@ -643,8 +643,8 @@ implementation more permissive about what operations are allowed::
       def __torch_function__(cls, func, types, args=(), kwargs=None):
           if kwargs is None:
               kwargs = {}
-          args = [a._t if hasattr(a, '_t') else a for a in args]
           metadatas = tuple(a._metadata for a in args if hasattr(a, '_metadata'))
+          args = [a._t if hasattr(a, '_t') else a for a in args]
```

Comment:
I think it is ok to have any order here as finally the first element is taken to create another MetadataTensor:
```python
return MetadataTensor(ret, metadata=metadatas[0])
```

Reply:
Sure, but I'm confused by the comment in https://github.com/pytorch/pytorch/pull/78073#issue-1244677756
I'm not a python syntax expert tbh. I always get that one wrong, but @pmeier seems to imply that one is correct and one is wrong. Since we use both (one for args and one for metadatas) then one has to be wrong? :p

Reply:
There are 2 possibilities in python:
1) inline if/else : `a = 1 if condition else 0` like C++ `auto a = (condition) ? 1.0 : 0.0`
2) conditional for-loop: `[i for i in range(10) if i % 2 == 0] == [0, 2, 4, ...]`

So, `args = [a._t if hasattr(a, '_t') else a for a in args]` constructs a list of `a._t` or `a` without skipping anything.
`metadatas = tuple(a._metadata for a in args if hasattr(a, '_metadata'))` creates a tuple of `a._metadata` using conditional for-loop.
Both syntaxes are OK now. The issue in #78073 is that syntax for inline if/else was incorrect and Philip replaced it by conditional for-loop.

This PR fixes another problem where `args` becomes overwritten and thus we can not construct correctly `metadatas`.